js/modules/k6/http/request.go

```diff
@@ -104,9 +104,13 @@ func (h *HTTP) Request(ctx context.Context, method string, url goja.Value, args
 		params = args[1]
 	}
 
-	req, err := h.parseRequest(ctx, method, u, body, params)
+	req, err := h.parseRequest(ctx, method, url, body, params)
 	if err != nil {
-		return nil, err
+		if state.Options.Throw.Bool {
+			return nil, err
+		}
+		state.Logger.WithField("error", err).Warn("Request Failed")
+		return &Response{Response: &httpext.Response{}}, nil
```

Comment:
I am for this to be created from some function and to include *more* stuff as currently `response.json()` ends up as a panic instead of simple(r) exception, this will likely be true for everything. 
Also I would argue that this should still have `error` and `error_code` set accordingly to what the problem was so you can detect what actually got wrong in the code.

Reply:
:-/ I still think it's strange to work with a `Response` object if a request was never made. IMO the correct behavior in this case would be to simply throw the parse exception and let the user handle it with `try`/`except` as usual, in which case they'd have access to the error object. But since the Throw option is so old and we don't promote `try`/`except` at all (it's nowhere in our docs :disappointed:) we've backed ourselves into this corner where we have to return a fake-valid `Response` object. Would it be too late to bring up the discussion of deprecating Throw?

I'll see how I can improve things so that at least there are no panics, but this feels way off.

Reply:
Yeah... :disappointed: This is definitely one of the things that we have to improve in the new HTTP API... Not sure exactly how we should improve it, considering the context and requirements of load testing, but it has so many corner cases and footguns... :disappointed:

Reply:
For me, this is a requirement for the merging of this PR